v2/devdoc/log_context_property_type_ascii_char_ptr_requirements.md

```diff
@@ -0,0 +1,95 @@
+
+Note: the copying of the string can be somewhat optimized by not using `snprintf`, this will be considered in a separate task.
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_004: [** `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string` shall succeed and return the result of `snprintf`. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_005: [** If any error is encountered, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string` shall fail and return a negative value. **]**
+
+## LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).copy
+
+```c
+typedef int (*LOG_CONTEXT_PROPERTY_TYPE_COPY)(void* dst_value, void* src_value);
+```
+
+`LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).copy` copies the `ascii_char_ptr` value from the address pointed by `src_value` to `dst_value`.
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_006: [** If `src_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).copy` shall fail and return a non-zero value. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_007: [** If `dst_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).copy` shall fail and return a non-zero value. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_008: [** `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).copy` shall copy the entire string (including the `NULL` terminator) from `src_value` to `dst_value`. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_009: [** `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).copy` shall succeed and return 0. **]**
+
+## LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).free
+
+```c
+typedef int (*LOG_CONTEXT_PROPERTY_TYPE_FREE)(void* value);
+```
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_018: [** `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).free` shall return. **]**
+
+## LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).get_type
+
+```c
+typedef LOG_CONTEXT_PROPERTY_TYPE (*LOG_CONTEXT_PROPERTY_TYPE_GET_TYPE)(void);
+```
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_010: [** `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).get_type` shall return the property type `LOG_CONTEXT_PROPERTY_TYPE_ascii_char_ptr`. **]**
+
+## LOG_CONTEXT_PROPERTY_TYPE_INIT(ascii_char_ptr)
+
+```c
+int LOG_CONTEXT_PROPERTY_TYPE_INIT(ascii_char_ptr)(void* dst_value, const char* format, ...);
+```
+
+`LOG_CONTEXT_PROPERTY_TYPE_INIT(ascii_char_ptr)` initializes the memory at `dst_value` with the `printf` style formatted string given by `format` and the arguments in `...`.
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_011: [** If `dst_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_INIT(ascii_char_ptr)` shall fail and return a non-zero value. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_012: [** `LOG_CONTEXT_PROPERTY_TYPE_INIT(ascii_char_ptr)` shall initialize the memory at `dst_value` with the `printf` style formatted string given by `format` and the arguments in `...`. **]**
```

Comment:
>initialize [](http://example.com/codeflow?start=114&length=10)

Can this fail in any way?

Reply:
Yes, can fail and SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_014 would kick in at that time.